panther_battery/src/adc_node.py

```diff
@@ -186,6 +193,25 @@ def _publish_battery_msg(
         else:
             battery_msg.power_supply_status = BatteryState.POWER_SUPPLY_STATUS_DISCHARGING
 
+        # check battery health
+        with self._lock:
+            if V_bat < self.V_BAT_FATAL_MIN:
+                battery_msg.power_supply_health = BatteryState.POWER_SUPPLY_HEALTH_DEAD
+                rospy.logerr(f'[{rospy.get_name()}] battery voltage critically low!')
+            elif battery_msg.percentage > 1.1:
```

Comment:
I would rather look at voltage. If the value exceeds 45 volts, this should mean overvoltage. By this should be after moving average